apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx

```diff
@@ -53,168 +54,210 @@ const prettyTimestamp = (timestamp: number) => {
   return `${day} ${month} ${year}`;
 };
 
-export default async function Proposal({
+export default function Proposal({
   params: { proposalId, poolId, chain, garden },
 }: {
   params: { proposalId: string; poolId: number; chain: number; garden: string };
 }) {
   // TODO: fetch garden decimals in query
-  const { data: getProposalQuery } = await queryByChain<getProposalDataQuery>(
-    urqlClient,
-    chain,
-    getProposalDataDocument,
-    {
-      garden: garden,
-      proposalId: proposalId,
-    },
-  );
+  const { data: getProposalQuery } =
+    useSubgraphQueryByChain<getProposalDataQuery>(
+      chain,
+      getProposalDataDocument,
+      {
+        garden: garden,
+        proposalId: proposalId,
+      },
+      {},
+      { topic: "proposal", id: proposalId, type: "update", chainId: chain },
+    );
 
   const proposalData = getProposalQuery?.cvproposal;
 
-  if (!proposalData) {
-    return (
-      <p className="text-center text-2xl text-error">{`Proposal ${proposalId} not found`}</p>
-    );
-  }
-
   const tokenSymbol = getProposalQuery?.tokenGarden?.symbol;
   const tokenDecimals = getProposalQuery?.tokenGarden?.decimals;
-  const proposalIdNumber = proposalData.proposalNumber as number;
-  const convictionLast = proposalData.convictionLast as string;
-  const threshold = proposalData.threshold as bigint;
-  const proposalType = proposalData.strategy.config?.proposalType as number;
-  const requestedAmount = proposalData.requestedAmount as bigint;
-  const beneficiary = proposalData.beneficiary as Address;
-  const submitter = proposalData.submitter as Address;
-  const status = proposalData.proposalStatus as number;
-  const stakedAmount = proposalData.stakedAmount as bigint;
-  const metadata = proposalData.metadata;
+  const proposalIdNumber = proposalData?.proposalNumber as number;
+  const convictionLast = proposalData?.convictionLast as string;
+  const threshold = proposalData?.threshold as bigint;
+  const proposalType = proposalData?.strategy.config?.proposalType as number;
+  const requestedAmount = proposalData?.requestedAmount as bigint;
+  const beneficiary = proposalData?.beneficiary as Address;
+  const submitter = proposalData?.submitter as Address;
+  const status = proposalData?.proposalStatus as number;
+  const stakedAmount = proposalData?.stakedAmount as bigint;
+  const metadata = proposalData?.metadata;
```

Comment:
forcing cast to be not undefined, if `proposalData` its undefined all the variables will also undefined.

That can potentially break the code that use that variables because they dont expect undefined.

Reply:
In theory it wont break, here is a part i have stashed some refactor but i can do it in the same pr if you want

Reply:
just feel wrong, if that was only one or two, but are several variables hard to guarantee that theory, not priority tho, but will have it in mind to improve.

Reply:
Yeha thats why I have somethiing locally that wraps all inside a single object

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/7de7a961e5d2a1ad87b6119713f6e277cab5ef5c